Fix Conversion suite narrowing conversion errors

Enabling narrowing errors by removing 
`if(CMAKE_COMPILER_IS_GNUCC OR "${CMAKE_CXX_COMPILER_ID}" MATCHES "(Apple)?Clang")
    add_cxx_flag_if_supported(-Wno-narrowing)
endif()` 
in test_conformance/conversions/CMakeLists.txt produces many errors. A common example is:
`test_conformance/conversions/basic_test_conversions.cpp:1624:5: error: narrowing conversion of '-125' from 'int' to 'unsigned int' inside { } [-Wnarrowing]`

--- a/test_conformance/conversions/CMakeLists.txt
+++ b/test_conformance/conversions/CMakeLists.txt
@@ -1,9 +1,5 @@
 set(MODULE_NAME CONVERSIONS)
 
-if(CMAKE_COMPILER_IS_GNUCC OR "${CMAKE_CXX_COMPILER_ID}" MATCHES "(Apple)?Clang")
-    add_cxx_flag_if_supported(-Wno-narrowing)
-endif()
-
 set (${MODULE_NAME}_SOURCES
       Sleep.cpp test_conversions.cpp basic_test_conversions.cpp
 )

